src/pages/CreateGroup/CreateGroup.tsx

```diff
@@ -0,0 +1,153 @@
+
+  const [groupName, setGroupName] = useState("");
+  const [curUserAdd, setCurUserAdd] = useState("");
+  const [groupUsers, setGroupUsers] = useState<SimpleUser[]>([]);
+
+  const [nameError, setNameError] = useState("");
+  const [userAddError, setUserAddError] = useState("");
+
+  // Create group, invite users, navigate to group, and show toast
+  const createGroup = async () => {
+    const newGroup = await api.post<Group>("/group/", {
+      name: groupName,
+    });
+    await api.put<Group>("/group/" + newGroup.data.id, {
+      invite_users: groupUsers.map((u) => u.username),
+    });
+    navigate("/groups/" + newGroup.data.id);
+    toast.custom(
+      <div className="flex w-[400px] justify-center rounded-lg bg-light_yellow p-4">
+        <p className="p-0 font-semibold text-secondary">
+          Successfully Created Group
+        </p>
+      </div>,
+    );
+  };
+
+  // If group name provided, open creation confirmation dialog
+  const validateName = () => {
+    if (!groupName) setNameError("Name cannot be empty.");
+    else {
+      window.HSOverlay.open(document.querySelector("#create-confirm")!);
+    }
+  };
+
+  // Adds user to invite list if input is non-empty, user is not already in group, and user is valid
+  const addUser = () => {
+    if (!curUserAdd) {
+      setUserAddError("Username cannot be empty.");
+    } else if (
+      groupUsers.findIndex(
+        (u) => u.username.toLowerCase() === curUserAdd.toLowerCase(),
+      ) !== -1 ||
+      curUserAdd.toLowerCase() === user!.username.toLowerCase()
+    ) {
+      setUserAddError("User is already a member.");
+    } else {
+      api
+        .get<User>("/user/username/" + curUserAdd)
+        .then((res) => {
+          setGroupUsers([...groupUsers, res.data]);
+          setUserAddError("");
+          setCurUserAdd("");
+        })
+        .catch(() => setUserAddError("User does not exist."));
+    }
+  };
+
+  // Removes user from invite list
+  const removeUser = (username: string) => {
+    setGroupUsers(groupUsers.filter((user) => user.username !== username));
+  };
+
+  return (
+    <>
+      <Column className="mx-8 mb-4 mt-16 gap-y-6">
+        {/* Header */}
+        <Row>
+          <Link to="/groups">
+            <Row>
+              <img src={LeftArrowIcon} className="max-w-[24px]" />
+              <label className="font-normal text-gray-600">Back</label>
+            </Row>
+          </Link>
+        </Row>
+        <h2 className="font-semibold">Create a Group</h2>
+        {/* Input Fields */}
+        <TextField
+          value={groupName}
+          onChangeValue={setGroupName}
+          label="Group Name:"
+          supportingText={nameError}
+          hasError={nameError !== ""}
+        />
+        <TextField
+          value={curUserAdd}
+          onChangeValue={setCurUserAdd}
+          label="Invite Users:"
+          supportingText={userAddError}
+          hasError={userAddError !== ""}
+          onEnterPressed={addUser}
```

Comment:
Could we have some sort of tooltip that tells user they need to press enter to add a user to the group they are creating?